core/src/replay_stage.rs

```diff
@@ -1446,7 +1555,11 @@ impl ReplayStage {
                     .clone();
 
                 // Signal retransmit
-                if Self::should_retransmit(poh_slot, &mut skipped_slots_info.last_retransmit_slot) {
+                if Self::should_retransmit(poh_slot, &mut skipped_slots_info.last_retransmit_slot)
+                    && retransmitted_slot
+                        .map(|slot| slot != bank.slot())
+                        .unwrap_or(true)
+                {
```

Comment:
Because the replay loop runs once every 100ms, it's probably unlikely that we'll get much savings out of this check (you would have to conveniently be doing the time-based retransmitting in the same iteration as this start leader check). 

Maybe we just pass in the `LastRetransmitInfo` here and check that a retransmit of `bank.slot()` has not occurred in the last `X` seconds?

Reply:
Also do we want to update `LastRetransmitInfo` if a retransmit occurs here? I.e. it counts towards the exponential backoff?

Reply:
good call, updated